core/Validators/NumberRange.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+
+namespace Piwik\Validators;
+
+use Piwik\Piwik;
+
+class NumberRange extends BaseValidator
+{
+    const MAX_SMALL_INT_UNSIGNED = 65535;
+    const MAX_MEDIUM_INT_UNSIGNED = 16777215;
+
+    /**
+     * @var null|int
+     */
+    private $min;
+
+    /**
+     * @var null|int
+     */
+    private $max;
+
+    /**
+     * @param null|int $min
+     * @param null|int $max
+     */
+    public function __construct($min = null, $max = null)
+    {
+        $this->min = $min;
+        $this->max = $max;
+    }
+
+    public function validate($value)
+    {
+        if ($value === false || $value === null || $value === '') {
+            // we allow this value. if it is supposed to be not empty, please use NotEmpty validator on top
+            return;
+        }
```

Comment:
Maybe it would be a good idea to move this to a protected member in BaseValidator? Then the comment could be on that method, and this code could look like:

```
if ($this->isValueEmpty($value)) {
    return;
}
```

Reply:
Good point, wonder what wording to use instead of "empty" as I would mean `empty()`. `isset()` is wrong as well. `isValueEmtpyButNotZero`?

Reply:
I'll go with `isValueBare ()` 👍